ankihub/main/importing.py

```diff
@@ -187,7 +190,106 @@ def import_ankihub_deck(
 
     def _import_note_types(self, note_types: Dict[NotetypeId, NotetypeDict]) -> None:
         self._import_note_types_into_ankihub_db(note_types=note_types)
-        _adjust_note_types_in_anki_db(note_types)
+        self._adjust_note_types_in_anki_db(note_types)
+
+    def _adjust_note_types_in_anki_db(
+        self, remote_note_types: Dict[NotetypeId, NotetypeDict]
+    ) -> None:
+        # can be called when installing a deck for the first time and when synchronizing with AnkiHub
+
+        LOGGER.info("Beginning adjusting note types...")
+        _create_missing_note_types(remote_note_types)
+        _rename_note_types(remote_note_types)
+        self._ensure_local_and_remote_fields_are_same(remote_note_types)
+        self._update_templates_and_css(remote_note_types)
+
+        LOGGER.info("Adjusted note types.")
+
+    def _ensure_local_and_remote_fields_are_same(
+        self, remote_note_types: Dict[NotetypeId, NotetypeDict]
+    ) -> None:
+        def field_tuples(flds: List[Dict]) -> List[Tuple[int, str]]:
+            return [(field["ord"], field["name"]) for field in flds]
+
+        note_types_with_field_conflicts: List[Tuple[NotetypeDict, NotetypeDict]] = []
+        for mid, remote_note_type in remote_note_types.items():
+            local_note_type = aqt.mw.col.models.get(mid)
+
+            # TODO Allow extra local fields
+            if not field_tuples(local_note_type["flds"]) == field_tuples(
```

Comment:
This got me thinking: we probably should change `add_note_type_fields()` to avoid changing ordinals of existing fields if we don't support that.

Reply:
I think we could actually remove the need for the ordinals to match and just require the remote field names to exist locally.
Maybe we could do that in https://ankihub.atlassian.net/jira/software/c/projects/BUILD/boards/1?selectedIssue=BUILD-1059?

About `add_note_type_fields`, it seems to me like there could be an issue with the ordinals, because the function first updates the ordinals for all fields based on the original note type, and then it later adjusts the special field’s ordinal by moving it to the end.